src/Parallel/CharmRegistration.hpp

```diff
@@ -153,12 +153,13 @@ struct RegisterParallelComponent : RegistrationHelper {
 
   void register_with_charm() const noexcept override {
     static bool done_registration{false};
+    static std::string parallel_component_name{};
     if (done_registration) {
       return;  // LCOV_EXCL_LINE
     }
     done_registration = true;
-    ckindex::__register(get_template_parameters_as_string<algorithm>().c_str(),
-                        sizeof(algorithm));
+    parallel_component_name = name();
```

Comment:
can't you define and set the `parallel_component_name` here? `static std::string ...`?

Reply:
Done, with `const` as well